catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/event/SendEvent.java

```diff
@@ -0,0 +1,253 @@
+        try (CloseableHttpClient httpClient = getClosableHttpClient()) {
+            List<Result> results = Arrays.asList(metacards)
+                    .stream()
+                    .map(ResultImpl::new)
+                    .collect(Collectors.toList());
+
+            SourceResponse queryResponse = new QueryResponseImpl(query,
+                    results,
+                    true,
+                    metacards.length);
+
+            LOGGER.debug("Attempting to transform an Event with mime-type: {} & outputSchema: {}",
+                    mimeType,
+                    outputSchema);
+            Map<String, Serializable> arguments = new HashMap<>();
+            if (elementName != null) {
+                arguments.put(CswConstants.ELEMENT_NAMES, elementName.toArray());
+            }
+
+            BinaryContent binaryContent = null;
+            if (!CswConstants.CSW_OUTPUT_SCHEMA.equals(outputSchema)) {
+                throw new WebApplicationException(new CatalogTransformerException(
+                        "Unable to locate Transformer."));
+            }
+            arguments.put(CswConstants.OUTPUT_SCHEMA_PARAMETER, outputSchema);
+            arguments.put(CswConstants.ELEMENT_SET_TYPE, elementSetType);
+            arguments.put(CswConstants.IS_BY_ID_QUERY, false);
+            arguments.put(CswConstants.GET_RECORDS, request);
+            arguments.put(CswConstants.RESULT_TYPE_PARAMETER, resultType);
+            arguments.put(CswConstants.WRITE_NAMESPACES, false);
+            QueryResponseTransformer transformer = transformerManager.getTransformerBySchema(
+                    outputSchema);
+            binaryContent = transformer.transform(queryResponse, arguments);
+
+            if (binaryContent == null) {
+                throw new WebApplicationException(new CatalogTransformerException(
+                        "Transformer returned null."));
+            }
+
+            HttpEntityEnclosingRequestBase httpMethod = getHttpRequest(operation);
+            httpMethod.addHeader("Cache-Control", "no-cache, no-store");
+            httpMethod.addHeader("Pragma", "no-cache");
+            httpMethod.setEntity(new ByteArrayEntity(binaryContent.getByteArray()));
+            ResponseHandler<String> responseHandler = new BasicResponseHandler();
+            return httpClient.execute(httpMethod, responseHandler);
+        } catch (Exception e) {
+            LOGGER.error("Error sending event to {}.", callbackUrl, e);
+        }
+        return null;
+    }
+
+    private HttpEntityEnclosingRequestBase getHttpRequest(Operation operation) {
+        HttpEntityEnclosingRequestBase httpMethod;
+        switch (operation) {
+        case UPDATE:
+            httpMethod = new HttpPut(callbackUrl);
+            break;
+        case DELETE:
+            httpMethod = new HttpDeleteWithBody(callbackUrl);
+            break;
+        default:
+            httpMethod = new HttpPost(callbackUrl);
```

Comment:
Nvm, just realized it was your own enum.

Instead of having this switch statement here, it almost feels like `Operation` should have a `getHttpRequest` method and be sub-classed. But that might be overkill...